src/Search/SearchHandler.php

```diff
@@ -103,7 +104,18 @@ public function search(AdminInterface $admin, $term, $page = 0, $offset = 20)
 
                 // NEXT_MAJOR: Remove this line.
                 $filter->setOption('case_sensitive', $this->caseSensitive);
-                $filter->setOption('or_group', $admin->getCode());
+
+                if ($filter instanceof ChainableFilterInterface) {
```

Comment:
I Think the SearchableFilterInterface should implements ChainableFilterInterface like I did in https://github.com/sonata-project/SonataAdminBundle/pull/7308